src/main/java/com/uber/cadence/internal/replay/WorkflowContext.java

```diff
@@ -166,7 +167,18 @@ SearchAttributes getSearchAttributes() {
 
     Map<String, Object> contextData = new HashMap<>();
     for (ContextPropagator propagator : contextPropagators) {
-      contextData.put(propagator.getName(), propagator.deserializeContext(headerData));
+      // Only send the context propagator the fields that belong to them
+      // Change the map from MyPropagator:foo -> bar to foo -> bar
+      Map<String, byte[]> filteredData =
+          headerData
+              .entrySet()
+              .stream()
+              .filter(e -> e.getKey().startsWith(propagator.getName()))
+              .collect(
+                  Collectors.toMap(
+                      e -> e.getKey().substring(propagator.getName().length() + 1),
+                      Map.Entry::getValue));
+      contextData.put(propagator.getName(), propagator.deserializeContext(filteredData));
```

Comment:
This is nice. Looks like the existing one is having issues with more than one ContextPropagators, right? @meiliang86 @mkolodezny 

However, I believe there is a backward-compatibility issue by doing it. 
The existing workflow using ContextPropagator already made headers without this format...
To be backward-compatible, maybe we should add an option to use this and default to false.

Reply:
Another idea is that we can safely assume the existing propagators don't start with some special string as key(e.g. `__cadence_v1_ctx_key` ) and then we can check if the headers have any keys with it. 

If yes then we can use this new way. If not, then fallback to use the old way.